pkg/api/api/grpc_status.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2025 The Bucketeer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"errors"
+
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	pkgErr "github.com/bucketeer-io/bucketeer/pkg/error"
+)
+
+const (
+	bktDomain = ".bucketeer.io"
+)
+
+func NewGRPCStatus(err error) *status.Status {
+	var bucketeerErr *pkgErr.BktError
+	var bktInvalidError *pkgErr.BktInvalidError
+	var bktFieldError *pkgErr.BktFieldError
+	if err == nil {
+		return status.New(codes.Unknown, "")
+	}
+	if errors.As(err, &bucketeerErr) {
+		return convertBktError(bucketeerErr)
+	} else if errors.As(err, &bktFieldError) {
+		return convertFieldError(bktFieldError)
+	} else if errors.As(err, &bktInvalidError) {
+		return convertInvalidError(bktInvalidError)
+	} else {
+		reason := "UNKNOWN"
+		st := status.New(codes.Unknown, err.Error())
+		metadata := map[string]string{
+			"message": err.Error(),
+		}
+		packageName := "unknown"
+
+		st, err = st.WithDetails(&errdetails.ErrorInfo{
+			Reason:   reason,
+			Domain:   packageName + bktDomain,
+			Metadata: metadata,
+		})
+		if err != nil {
+			return status.New(codes.Internal, err.Error())
+		}
+		return st
+	}
+}
```

Comment:
BktFieldError and BktInvalidError embed *BktError, so they will always match the first condition and never reach their specific handlers.

```suggestion
func NewGRPCStatus(err error) *status.Status {
    if err == nil {
        return status.New(codes.Unknown, "")
    }
    
    // Check most specific types first
    var bktInvalidError *pkgErr.BktInvalidError
    var bktFieldError *pkgErr.BktFieldError  
    var bucketeerErr *pkgErr.BktError
    
    if errors.As(err, &bktInvalidError) {
        return convertInvalidError(bktInvalidError)
    } else if errors.As(err, &bktFieldError) {
        return convertFieldError(bktFieldError)
    } else if errors.As(err, &bucketeerErr) {
        return convertBktError(bucketeerErr)
    } else {
        return convertUnknownError(err)
    }
}
```